lib/site_prism/element_container.rb

```diff
@@ -71,6 +71,20 @@ def raise_if_block(obj, name, has_block)
       raise SitePrism::UnsupportedBlock, "#{obj.class}##{name}"
     end
 
+    def raise_wait_for_if_failed(obj, name, timeout, failed)
```

Comment:
These methods become part of the public API.

Reply:
Whilst I agree with your premise. He's copied the existing structure.

This entire area needs some rework. I'd rather get this PR in as is (I've got some additional Items I'm going to be adding to the TODO document). If you're ok with merging this with this caveat I'll merge this in. I'd rather have you're agreement though.